bin/tidyall

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env perl
```

Comment:
`tidyall` is provides by the perl module Code::TidyAll, so I think creating a script with the same name is unfortunate.
Like I said, tidyall provides tidy tools for several languages/tools, including perltidy.

We use it mainly because it has a caching functionality, so it is faster to use locally.
In CI we are using https://github.com/os-autoinst/openQA/blob/master/t/00-tidy.t and https://github.com/os-autoinst/os-autoinst/blob/master/xt/00-tidy.t (and yes, it's unfortunate that they are different :)

Reply:
I'm very open to suggestions on naming, the goal with this is to minimize scattering across repos and differences.

from your examples I see two ways of approaching linter checks:

- Run it as a test in `t/` (which I guess `prove` runs).
- Run it as a standalone command (which is what the `openQA` version is doing).

I simplified the latter, as I see there are may versions of the same script. But I don't know if either of the versions is more or less beneficial.

I guess it's also team agreement on which strategy to pick.

IMHO: I lean for the standalone command because it looks more similar to how other stacks work:

+ in Ruby you'd install dependencies & `bundle exec rubocop`
+ in Python you'd install dependencies, virtualenv [with your favourite virtualenv tool] & `{black|flake8|pylint}`.
+ in JS you install dependencies & `eslint`.

Reply:
But I'm also wondering about other things.
I use `tools/tidy -o` to only tidy changed files.
Also it has a check-only mode.

And tools for development shouldn't be in `bin/`, as this is usually the directory that contains scripts that are packaged.